HEADER.js

```diff
@@ -71,95 +77,25 @@ fabric.SHARED_ATTRIBUTES = [
 ];
 /* _FROM_SVG_END_ */
 
-/**
- * Pixel per Inch as a default value set to 96. Can be changed for more realistic conversion.
- */
-fabric.DPI = 96;
+
 fabric.reNonWord = /[ \n\.,;!\?\-]/;
-fabric.fontPaths = { };
-fabric.iMatrix = iMatrix;
 
 
-/**
- * Pixel limit for cache canvases. 1Mpx , 4Mpx should be fine.
- * @since 1.7.14
- * @type Number
- * @default
- */
-fabric.perfLimitSizeTotal = 2097152;
 
 /**
- * Pixel limit for cache canvases width or height. IE fixes the maximum at 5000
- * @since 1.7.14
- * @type Number
- * @default
+ * @todo move to config when window is exported
  */
-fabric.maxCacheSideLimit = 4096;
-
-/**
- * Lowest pixel limit for cache canvases, set at 256PX
- * @since 1.7.14
- * @type Number
- * @default
- */
-fabric.minCacheSideLimit = 256;
+config.configure({
+  devicePixelRatio: fabric.window.devicePixelRatio ||
+    fabric.window.webkitDevicePixelRatio ||
+    fabric.window.mozDevicePixelRatio ||
+    1
+});
 
 /**
  * Cache Object for widths of chars in text rendering.
  */
-fabric.charWidthsCache = { };
-
-/**
- * if webgl is enabled and available, textureSize will determine the size
- * of the canvas backend
- * @since 2.0.0
- * @type Number
- * @default
- */
-fabric.textureSize = 2048;
-
-/**
- * When 'true', style information is not retained when copy/pasting text, making
- * pasted text use destination style.
- * Defaults to 'false'.
- * @type Boolean
- * @default
- */
-fabric.disableStyleCopyPaste = false;
-
-/**
- * Enable webgl for filtering picture is available
- * A filtering backend will be initialized, this will both take memory and
- * time since a default 2048x2048 canvas will be created for the gl context
- * @since 2.0.0
- * @type Boolean
- * @default
- */
-fabric.enableGLFiltering = true;
-
-/**
- * Device Pixel Ratio
- * @see https://developer.apple.com/library/safari/documentation/AudioVideo/Conceptual/HTML-canvas-guide/SettingUptheCanvas/SettingUptheCanvas.html
- */
-fabric.devicePixelRatio = fabric.window.devicePixelRatio ||
-                          fabric.window.webkitDevicePixelRatio ||
-                          fabric.window.mozDevicePixelRatio ||
-                          1;
-/**
- * Browser-specific constant to adjust CanvasRenderingContext2D.shadowBlur value,
- * which is unitless and not rendered equally across browsers.
- *
- * Values that work quite well (as of October 2017) are:
- * - Chrome: 1.5
- * - Edge: 1.75
- * - Firefox: 0.9
- * - Safari: 0.95
- *
- * @since 2.0.0
- * @type Number
- * @default 1
- */
-fabric.browserShadowBlurConstant = 1;
+fabric.charWidthsCache = {};
```

Comment:
i m going to take care of this later for the charWidthsCache. I want to change it to a Map.

Reply:
excellent

but change it in #8198